### PR TITLE
Don't compile classes for Java 9

### PR DESCRIPTION
**Pull request by diesieben07**
We cannot compile for Java 9 (specifically not `NashornScriptEngineFactory`) as it will be loaded on Java 8 when looking up a script engine by name (e.g. using `new ScriptEngineManager().getEngineByName("JavaScript");`) as that loops through all engines discoverable via service loader.

### CODE REVIEW

**Comment by LexManos** [comment]
In my tests this is not an issue. It'll try and load it, error and move on.
The issue is the linker extension requires java 9 to compile as it only exists there.
We also don't expose ourselves as JavaScript, so it shouldn't be a issue.
What about moving the two classes into two source sets? One targeting the J8 toolchain and the other J9?

Do we have a real world issue report/reproduction for this? I'd like to verify on my end and figure out why my tests {which included asking the factory by name like you said} didn't trigger.

**Comment by diesieben07** [comment]
This was reported on the forums: https://forums.minecraftforge.net/topic/97014-1165-nashorn-script-engine-issue/?tab=comments#comment-441145
The code snippet I posted above will crash every time on Java 8. It doesn't matter that you don't expose it as Javascript, because to know that, the JVM needs to load that class and look at what it exposes itself as - which will already fail because its a Java 9 class on Java 8.

> What about moving the two classes into two source sets? One targeting the J8 toolchain and the other J9?

This sounds doable, I'll see what I can do.

**Comment by EwyBoy** [comment]
Hey, ran into same issue as @OreCruncher did on this forum thread: 
https://forums.minecraftforge.net/topic/97014-1165-nashorn-script-engine-issue/?tab=comments#comment-441145
and while attempting to solve this issue I ended up here.

When I was trying to execute the line:
```java
ScriptEngine engine = new ScriptEngineManager().getEngineByName("Nashorn");
```
I got the same error with the Java version target as mentioned above.

I just wanted to inform anyone who happens to stumble upon this thread just like me, that you can temporary resolve this issue by setting the loader parameter to `null` while creating a new instance of a `ScriptEngineManager`.

TLDR:
```java
This crashes: ScriptEngine engine = new ScriptEngineManager().getEngineByName("Nashorn");
This doesnt:  ScriptEngine engine = new ScriptEngineManager(null).getEngineByName("Nashorn");
```

**Comment by LexManos** [comment]
Seems Diesieben hasnt gotten around to updating this to do it properly.
So guess we can pull for the time being, but it really should be split into two sourcesets compiled against the proper toolchains. And the version should be determined by the commits since tag like all our other projects. But that's work that isnt really needed for something this small. As this was only meant to be a small script to do what we needed.
